ArjunaJTS/jts/classes/com/arjuna/ats/jts/utils/Utility.java

```diff
@@ -258,6 +258,9 @@ public static final Uid getUid (org.omg.CosTransactions.Control cont)
 
 	byte[] nodeName = TxControl.getXANodeName().getBytes(StandardCharsets.UTF_8);
 
+        if (Boolean.getBoolean("GLASSFISH-21532-WORKAROUND"))
```

Comment:
This workaround is only required for recovery. I plan to raise a separate JIRA for any recovery related fixes so this needs removing.